### PR TITLE
Fix printf format warnings

### PR DESCRIPTION
**Pull request by yamt**
## Summary
fix some printf errors found by the compiler.
## Testing
https://github.com/apache/incubator-nuttx/pull/2222

### CODE REVIEW

File: system/ramtest/ramtest.c [hunk 1]
```diff
@@ -142,8 +142,8 @@ static void parse_commandline(int argc, char **argv,
   info->start = (uintptr_t)strtoul(argv[optind], &ptr, 16);
   if (*ptr != '\0')
     {
-      printf(RAMTEST_PREFIX "Invalid <hex-address>: %s->%lx [%02x]\n",
-             argv[optind], info->start, *ptr);
+      printf(RAMTEST_PREFIX "Invalid <hex-address>: %s->%p [%02x]\n",
```
Comment (xiaoxiang781216) [inline system/ramtest/ramtest.c:145]: should we change p to PRIxPTR?

File: nshlib/nsh_dbgcmds.c [hunk 2]
```diff
@@ -445,7 +445,8 @@ int cmd_hexdump(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
                       dumpbytes = count;
                     }
 
-                  snprintf(msg, sizeof(msg), "%s at %08x", argv[1], skip);
+                  snprintf(msg, sizeof(msg), "%s at %08zx", argv[1],
+                           (uintmax_t)skip);
```
Comment (xiaoxiang781216) [inline nshlib/nsh_dbgcmds.c:449]: should we use ssize_t?

File: nshlib/nsh_dbgcmds.c [hunk 3]
```diff
@@ -473,7 +474,8 @@ int cmd_hexdump(FAR struct nsh_vtbl_s *vtbl, int argc, FAR char **argv)
             }
 #endif
 
-          snprintf(msg, sizeof(msg), "%s at %08x", argv[1], position);
+          snprintf(msg, sizeof(msg), "%s at %08zx", argv[1],
+                   (uintmax_t)position);
```
Comment (xiaoxiang781216) [inline nshlib/nsh_dbgcmds.c:478]: should we use ssize_t?

File: system/ramtest/ramtest.c (hunk 1 above)
Comment (yamt) [reply system/ramtest/ramtest.c]: done

File: nshlib/nsh_dbgcmds.c (hunk 2 above)
Comment (yamt) [reply nshlib/nsh_dbgcmds.c:449]: i changed it to use uintmax_t consistently.

File: nshlib/nsh_dbgcmds.c (hunk 3 above)
Comment (yamt) [reply nshlib/nsh_dbgcmds.c:478]: i changed it to use uintmax_t consistently.

File: nshlib/nsh_dbgcmds.c (hunk 2 above)
Comment (xiaoxiang781216) [reply nshlib/nsh_dbgcmds.c:449]: consistent with what?

Comment (yamt) [reply nshlib/nsh_dbgcmds.c:449]: i made printf format (%jx) and cast (uintmax_t) consistent.

ssize_t might or might not be wider than off_t.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**